### ISSUE TITLE
Add an optional focus ring to `VLink` component

### ISSUE THREAD

**Issue by obulat** [WordPress/openverse#479]
## Description
Some links require focus ring. To ensure that the focus rings are always consistent, we should add a `focus-ring` prop to `VLink`, and add the necessary tailwind classes to add a focus ring if the prop is true. By default, the prop should be false.

## Note
Based on the discussion in WordPress/openverse-frontend#2015: 
    > I don't think we can set a fixed height in the navigation bar because we need the designs to be used in other languages

Very much agree with this. My point was about the need for `8px` of padding as the focus style looks strange.

<img width="353" alt="CleanShot 2022-12-07 at 19 16 29@2x" src="https://user-images.githubusercontent.com/895819/206263469-26ef19ce-13bf-41ec-a1e7-8b1ecee426d8.png">

Your point about the tappable area makes sense, but I think we don't need it as the [Target Size Criterion](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) requires _at least_ a `44px` area.

What you did is definitely correct. I am just bringing this up to reach consistency across other similar elements. I am thinking of the "Clear" action in filters, "Report this content" in the single result, these nav links in the Footer, and possibly others that I am missing.

The `Clear` component does exist in Figma, but since it does not in code, and we are discussing its purpose, it seems timely to define a basic link element that does not have paddings/margins, could have an icon, and where focus ring wraps the element. Something like the image below, but called `VSingleLink` instead (name TBD).

<img width="300" alt="CleanShot 2022-12-07 at 19 24 54@2x" src="https://user-images.githubusercontent.com/895819/206265159-76b8e9eb-a6e3-4993-8948-dd59235c5708.png">

What do you think?

_Originally posted by @panchovm in https://github.com/WordPress/openverse-frontend/issues/2015#issuecomment-1341424072_
